packages/expo-router/src/modal/Modal.tsx

```diff
@@ -0,0 +1,143 @@
+import { type NavigationProp, type ParamListBase } from '@react-navigation/native';
```

Comment:
Add "use client", render Modal in the __rsc_tests__ to ensure it can render on the server.